pkg/action/archive.go

```diff
@@ -140,6 +142,27 @@ func extractTar(ctx context.Context, d string, f string) error {
 		return nil
 	case strings.Contains(filename, ".bz2") || strings.Contains(filename, ".bzip2"):
 		br := bzip2.NewReader(tf)
+		// Extract non-tar archives directly
+		if !strings.HasSuffix(filename, ".tar.bz2") {
+			uncompressed := strings.TrimSuffix(filepath.Base(f), ".bz2")
+			uncompressed = strings.TrimSuffix(uncompressed, ".bzip2")
+			target := filepath.Join(d, uncompressed)
+			if err := os.MkdirAll(filepath.Dir(target), 0o755); err != nil {
+				return fmt.Errorf("failed to create directory for file: %w", err)
+			}
+
+			// #nosec G115
+			out, err := os.OpenFile(target, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0o600)
```

Comment:
I can't tell - but is this creating a new file in the same directory as the source file?  That may not work in all circumstances; for example, scanning a read-only mount or local directory where the user has read but not write access.

Reply:
It's landing in `/tmp` unless I'm misunderstanding:
```
time=2024-12-12T15:30:33.982-06:00 level=DEBUG source=.../repos/chainguard-dev/malcontent/pkg/action/archive.go:86 msg="extracting tar" dir=/var/folders/n6/xxn5d2zd3l1gghpx_7qppzs00000gn/T/test1.csv.bz21307193049
```